### PR TITLE
Beta release docs

### LINKED ISSUE #804: Beta Release Documentation
We will expand this. For now keeping notes on what is needed:
- Versioning breaking change documentation
- Uninstall instructions for pre-beta

### PR DESCRIPTION
**Pull request by blackdwarf**
Adding beta release docs as well as dependencies page for the CLI. Also
adding changes to the Known issues document for Beta.

Fix #804 

/cc @brthor @piotrpMSFT

### CODE REVIEW

File: Documentation/beta-release-notes.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+2. Delete the non-release version of the package
+
+# Uninstalling dotnet
+Due to the above breaking change, we reccommend that you uninstall the pre-Beta bits. Below are instructions for all of the supported platforms. 
+
+## Windows
+Simply uninstall the MSI using Control Panel. 
+
+## Ubuntu
+The best way to remove the package is to purge it with the following command: `sudo apt-get purge [package-name]`. This will remove all of the dotnet that was installed via apt-get. The `[[ackage-name]` above will most likely be `dotnet` but if you installed dev or nightly it will be `dotnet-dev` or `dotnet-nightly` respectivelly.
```
Comment (brthor) [inline Documentation/beta-release-notes.md]: typo: `[[ackage-name]`

File: Documentation/cli-dependencies.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+Native dependencies and requirements
+====================================
+
+As any other software on the planet, the CLI toolchain and the .NET Core framework that powers it (and is powered by it) have certain native dependencies across the platforms that it supports. These dependencies are listed in this document as well as the best way to install them on your machine. 
+
+# When do I need to worry about this?
+TBD
```
Comment (brthor) [inline Documentation/cli-dependencies.md]: ?

File: Documentation/beta-release-notes.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+* [Ubuntu](https://dotnetcli.blob.core.windows.net/dotnet/beta/Binaries/Latest/dotnet-ubuntu-x64.latest.tar.gz)
+* [CentOS](https://dotnetcli.blob.core.windows.net/dotnet/beta/Binaries/Latest/dotnet-centos-x64.latest.tar.gz)
+* [OS X](https://dotnetcli.blob.core.windows.net/dotnet/beta/Binaries/Latest/dotnet-osx-x64.latest.tar.gz)
+
+Installing is as simple as unzipping/untarring the file into a location.  After that, do not forget to define the `$DOTNET_HOME` environment variable to the directory where you extracted the archive (the one containing the `bin` and `runtime` directories). 
+
+Also, please refer to the following dependencies that you need for each platform, since the 
+
+## Building from source
+If you're adventureous and want to build from source, it is quite easy. Clone this repo and run `build.cmd` or `build.sh` for Windows and Linux/OS X respectively.
```
Comment (brthor) [inline Documentation/beta-release-notes.md]: typo: adventurous

File: Documentation/beta-release-notes.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
+.NET Command Line (CLI) tools - BETA Readme
+===========================================
+
+# Breaking changes
+There are two main breaking changes in beta. 
+
+1. Versioning change - we have changed the way we version the packages and introduced a new branching mechanism. You can see more in #668. 
+2. Nightly and dev packages cannot co-exist with release packages on one box.
```
Comment (brthor) [inline Documentation/beta-release-notes.md]: Nightly and dev packages are the same thing here

File: Documentation/beta-release-notes.md (hunk 1 above)
Comment (brthor) [inline Documentation/beta-release-notes.md]: What do you think about including just a blanket list people can copy/paste here.

e.g. To get the new bits:
```bash
sudo apt-get purge dotnet dotnet-dev dotnet-nightly
sudo apt-get update
sudo apt-get install dotnet
```

File: Documentation/beta-release-notes.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+## OS X 
+OS X doesn't provide any native uninstall option for PKG installs, so you will have to do some manual work. Luckily, it is not difficult if you follow the steps below:
+
+1. Remove the `/usr/local/share/dotnet` directory and everything in it.
+2. Remove all of the `dotnet*` symbolic links in `/usr/local/bin`. 
+3. Remove the following files, if they exist:
+    * `/usr/local/bin/corehost`
+    * `/usr/local/bin/libcorelcr.dylib`
+4. "Forget" the PKG recipe using the following command: `pkgutil --packages | grep dotnet | xargs pkgutil --forget`
```
Comment (brthor) [inline Documentation/beta-release-notes.md]: Same here, I think something people could copy/paste would make this easier.

```
rm -f /usr/local/share/dotnet
rm /usr/local/bin/dotnet*
rm /usr/local/bin/corehost
rm /usr/local/bin/libcoreclr.dylib
pkgutil --packages | grep dotnet | xargs pkgutil --forget
`

File: Documentation/beta-release-notes.md (hunk 1 above)
Comment (blackdwarf) [inline Documentation/beta-release-notes.md]: Not a bad idea. :)

File: Documentation/cli-dependencies.md (hunk 2 above)
Comment (blackdwarf) [inline Documentation/cli-dependencies.md]: Ups. :(

File: Documentation/beta-release-notes.md (hunk 5 above)
Comment (blackdwarf) [inline Documentation/beta-release-notes.md]: Here, I would rather not, since it is removing physical stuff.

**Comment by blackdwarf** [comment]
@brthor updated the PR.

File: Documentation/beta-release-notes.md (hunk 5 above)
Comment (brthor) [inline Documentation/beta-release-notes.md]: fair enough

**Comment by brthor** [comment]
Looks good to me :shipit:

File: Documentation/cli-dependencies.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
+Native dependencies and requirements
+====================================
+
+As any other software on the planet, the CLI toolchain and the .NET Core framework that powers it (and is powered by it) have certain native dependencies across the platforms that it supports. These dependencies are listed in this document as well as the best way to install them on your machine. 
+
+# Ubuntu
+The following dependencies are needed on Linux:
+
+* cmake
```
Comment (piotrpMSFT) [inline Documentation/cli-dependencies.md]: are these the names of the packages one should sudo apt-get install?

**Comment by piotrpMSFT** [comment]
We also need installer/zip URLs in this PR.

File: Documentation/cli-dependencies.md (hunk 6 above)
Comment (blackdwarf) [inline Documentation/cli-dependencies.md]: @piotrpMSFT  yep. There are ready copy-paste commands below.

**Comment by blackdwarf** [comment]
@piotrpMSFT beta-release-notes file has the links. Unless you mean links in the main README.md?

**Comment by piotrpMSFT** [comment]
not calling it beta anymore. Preview, right?

**Pull request closed without merging**